governance/program/src/state/proposal.rs

```diff
@@ -102,11 +110,182 @@ impl Proposal {
+            self.get_current_vote_threshold(self.yes_votes_count, governing_token_supply);
+
+        let final_state = if current_yes_vote_threshold > config.vote_threshold_percentage as u64 {
+            ProposalState::Succeeded
+        } else {
+            ProposalState::Defeated
+        };
+
+        self.state = final_state;
+        self.voting_completed_at = Some(current_slot);
+
+        Ok(())
+    }
+
+    /// Returns current vote threshold in relation to the total governing token supply
+    fn get_current_vote_threshold(&self, vote_count: u64, governing_token_supply: u64) -> u64 {
+        vote_count
+            .checked_mul(100)
+            .unwrap()
+            .checked_div(governing_token_supply)
+            .unwrap()
+    }
+
+    /// Checks if vote can be tipped and automatically transitioned to Succeeded or Defeated state
+    /// If the conditions are met the state is updated accordingly
+    pub fn try_tip_vote(
+        &mut self,
+        governing_token_supply: u64,
+        config: &GovernanceConfig,
+        current_slot: Slot,
+    ) -> Result<(), ProgramError> {
+        if let Some(tipped_state) = self.try_get_tipped_vote_state(governing_token_supply, config) {
+            self.state = tipped_state;
+            self.voting_completed_at = Some(current_slot);
+        }
+
+        Ok(())
+    }
+
+    /// Checks if vote can be tipped and automatically transitioned to Succeeded or Defeated state
+    /// If yes then Some(ProposalState) is returned and None otherwise
+    pub fn try_get_tipped_vote_state(
+        &self,
+        governing_token_supply: u64,
+        config: &GovernanceConfig,
+    ) -> Option<ProposalState> {
+        if self.yes_votes_count == governing_token_supply {
+            return Some(ProposalState::Succeeded);
+        }
+        if self.no_votes_count == governing_token_supply {
+            return Some(ProposalState::Defeated);
+        }
+
+        let current_yes_vote_threshold =
+            self.get_current_vote_threshold(self.yes_votes_count, governing_token_supply);
+
+        // We can only tip the vote automatically to Succeeded if more than 50% votes have been cast as Yeah
+        // and the Nay sayers can't change the outcome any longer
+        if current_yes_vote_threshold >= 50
+            && current_yes_vote_threshold > config.vote_threshold_percentage as u64
+        {
+            return Some(ProposalState::Succeeded);
+        } else {
+            let current_no_vote_threshold =
+                self.get_current_vote_threshold(self.no_votes_count, governing_token_supply);
+
+            // We can  tip the vote automatically to Defeated if more than 50% votes have been cast as Nay
+            // or the Yeah sayers can't outvote Nay any longer
+            // Note: Even splits  resolve to Defeated
+            if current_no_vote_threshold >= 50
+                || current_no_vote_threshold >= 100 - config.vote_threshold_percentage as u64
```

Comment:
If this is the second option presented earlier, this should probably be:
```suggestion
                && current_no_vote_threshold >= config.vote_threshold_percentage as u64
```

Reply:
I changed `vote_threshold_percentage` to `yes_vote_threshold_percentage` to disambiguate this scenario. With that change the No vote threshold is `100 - yes_vote_threshold_percentage`